libevmasm/AssemblyItem.cpp

```diff
@@ -168,7 +168,8 @@ size_t AssemblyItem::bytesRequired(size_t _addressLength, Precision _precision)
 size_t AssemblyItem::arguments() const
 {
 	if (type() == Operation)
-		return static_cast<size_t>(instructionInfo(instruction()).args);
+		//TODO:prevrandao: double check if we can use the default EVM version here
```

Comment:
Yeah, to simplify things, we can just require ``args`` and ``ret`` to be the same in all ``instructionInfo``s across all EVM versions - that'd also mean we don't need to pass the evm version around in some other places.
We'd just need to document that in some comments in ``instructionInfo`` to make sure we have another look in the unlikely event that this would ever be violated in the future.

Reply:
Would this assumption also extend to other `InstructionInfo` fields? Like `gasPriceTier` (used by `runGas`, for example)? If so, it would simplify even more the changes.

Reply:
@cameel do you have any input on this as well?

Reply:
Seems ok to do the same thing with most other places where it's viable. If the assumption stops being true, we can just add the evm version arg to those functions then.

Regarding specifically the ones you mentioned:
- `sideEffects`: IMO this is unlikely to change between versions.
- `gasPriceTier`: this is technically more likely (repricings aren't that rare) but this also seems to be a part of the gas estimator, which we're going to remove anyway. I don't think we'll need the parameter here before that happens.